### PR TITLE
[ticket/13598] Allow topic lock on topic creation

### PR DESCRIPTION
**Pull request by RMcGirr83**
Allow a user, if they have the permission, to lock a topic upon creation of a topic.

[PHPBB3-13598](https://tracker.phpbb.com/browse/PHPBB3-13598)

### CODE REVIEW

**Comment by RMcGirr83** [comment]
https://github.com/phpbb/phpbb/pull/3495

**Comment by RMcGirr83** [comment]
No idea why this is failing on Travis :roll:

**Comment by Elsensee** [comment]
You're changing `submit_post` so it requires a value for `'topic_status'` in the `$post_data` array. The tests don't know that yet and so they don't have a value for `'topic_status'` in their `$post_data` array. :wink:

**Comment by RMcGirr83** [comment]
Yeah, tests confuse me.  No idea on how to correct this.

**Comment by Elsensee** [comment]
Two possible ways:
1. change your.. change in `submit_post` to something like `isset($data['topic_status']) ? $data['topic_status'] : ITEM_UNLOCKED` - this also is kind of a  "backwards compatiblity" for some extensions which are using `submit_post` and would now have to change it so they don't throw that SQL error you're currently seeing in travis
2. change the tests so they have a value for `topic_status` in their `$post_data`.

I'd go for the first way because of the "backward compatiblity" thing. :grin:

**Comment by RMcGirr83** [comment]
Thanks @Elsensee 

"so they don't throw that SQL error you're currently seeing in travis"

I didn't see the SQL error...just some gibberish in a red box which I assumed was the error but it really didn't say much.

Ticket updated

File: phpBB/posting.php [hunk 1]
```diff
@@ -1758,7 +1764,7 @@
 	'S_SIGNATURE_CHECKED'		=> ($sig_checked) ? ' checked="checked"' : '',
 	'S_NOTIFY_ALLOWED'			=> (!$user->data['is_registered'] || ($mode == 'edit' && $user->data['user_id'] != $post_data['poster_id']) || !$config['allow_topic_notify'] || !$config['email_enable']) ? false : true,
 	'S_NOTIFY_CHECKED'			=> ($notify_checked) ? ' checked="checked"' : '',
-	'S_LOCK_TOPIC_ALLOWED'		=> (($mode == 'edit' || $mode == 'reply' || $mode == 'quote') && ($auth->acl_get('m_lock', $forum_id) || ($auth->acl_get('f_user_lock', $forum_id) && $user->data['is_registered'] && !empty($post_data['topic_poster']) && $user->data['user_id'] == $post_data['topic_poster'] && $post_data['topic_status'] == ITEM_UNLOCKED))) ? true : false,
+	'S_LOCK_TOPIC_ALLOWED'		=> (($mode == 'edit' || $mode == 'reply' || $mode == 'quote' || $mode == 'post') && ($auth->acl_get('m_lock', $forum_id) || ($auth->acl_get('f_user_lock', $forum_id) && $user->data['is_registered'] && !empty($post_data['topic_poster']) && $user->data['user_id'] == $post_data['topic_poster'] && $post_data['topic_status'] == ITEM_UNLOCKED))) ? true : false,
```
Comment (nickvergessen) [inline phpBB/posting.php]: What other modes do exist? can we drop the mode part now?

Comment (RMcGirr83) [inline phpBB/posting.php]: None that seem to have to do with posting/replying/etc..  Bump, delete, soft_delete, smilies, popup.  Can't see why those would matter though.

**Comment by RMcGirr83** [comment]
@nickvergessen do we have a consensus that those modes can now be removed?

File: phpBB/posting.php (hunk 1 above)
Comment (nickvergessen) [inline phpBB/posting.php]: Nevermind, seems okay

**Comment by marc1706** [comment]
I guess the PR is fine the way it looks right now.

**Comment by Nicofuma** [comment]
:+1:

**Comment by RMcGirr83** [comment]
:point_up:

**Pull request merged**